pages/index.js

```diff
@@ -20,14 +22,31 @@ const StyledLink = styled(Link)`
 `;
 
 export default function HomePage({ transactions }) {
+  const [search, setSearch] = useState({
+    name: "",
+    category: "",
+    direction: "",
+    dateFrom: "",
+    dateTo: "",
+    amountFrom: "",
+    amountTo: "",
+  });
+
+  const isSearchEntry = Object.values(search).some((value) => value !== "");
+
   return (
     <main>
       <Header title="Transactions" />
       <StyledDiv>
+        <SearchBar
+          transactions={transactions}
+          search={search}
+          onSearch={setSearch}
+          isSearchEntry={isSearchEntry}
+        />
         <StyledLink href="/newentry">Add New Transaction</StyledLink>
       </StyledDiv>
-
-      <TransactionList transactions={transactions} />
+      {isSearchEntry ? <></> : <TransactionList transactions={transactions} />}
```

Comment:
Here you can pass the filtered transactions to your list, no need for two lists.

Reply:
We have now changed this to the existing list